### PR TITLE
Change format keyword positioning

### LINKED ISSUE #187: Format is not only allowed for string values
At https://json-schema.org/understanding-json-schema/reference/string.html#format it says:
> The format keyword allows for basic semantic validation on certain kinds of **string** values that are commonly used. 

As discussed in issue https://github.com/json-schema-org/json-schema-spec/issues/759, format is also defined for non-string types. Therefore, the description should be made more general and moved to another place as it doesn't really fit where it is now. I guess it's better placed at the end of https://json-schema.org/understanding-json-schema/reference/type.html or so...

### PR DESCRIPTION
**Pull request by bhavukkalra**
**What kind of change does this PR introduce?**
PR moves the `format` section of the documentation from the misplaced [string](https://json-schema.org/understanding-json-schema/reference/string) page to under the more generic [type](https://json-schema.org/understanding-json-schema/reference/type) as discussed in [Issue #759](https://github.com/json-schema-org/json-schema-spec/issues/759). Which aims to remove the Misinterpretation that the reader might get that `format` as a keyword is only for string types which is not the case.

Also, Add further clarification that `format` as a keyword is generic. Using these lines added further

image

**Issue Number:**
-  Closes #187 

**Screenshots**

- Added section under [type](https://json-schema.org/understanding-json-schema/reference/type) page

image

- removed from `format` section from [string](https://json-schema.org/understanding-json-schema/reference/string) page

image

**Summary**
PR aims to re-adjust the positioning of `format` keyword definitions. To remove any confusion of it being a `string` specific keyword as reported by the users here [Slack Archives](https://json-schema.slack.com/archives/C5CF75URH/p1562596499404200)

**Does this PR introduce a breaking change?**
No

CC - @benjagm @gregsdennis @jviotti

### CODE REVIEW

**Comment by benjagm** [comment]
> misplaced [string](https://json-schema.org/understanding-json-schema/reference/string) page to under the more generic [type](https://json-schema.org/understanding-json-schema/reference/type) as discussed in https://github.com/json-schema-org/json-schema-spec/issues/759

I checked the issue https://github.com/json-schema-org/json-schema-spec/issues/759 and it was closed 2019. Are you sure you are referring to the right issue?

**Review by benjagm — APPROVED** [review]
Great work Bhavuk, Thanks!!

File: pages/understanding-json-schema/reference/type.md [hunk 1]
```diff
@@ -175,3 +175,129 @@ types. For example, numeric types have a way of specifying a numeric
 range, that would not be applicable to other types. In this reference,
 these validation keywords are described along with each of their
 corresponding types in the following chapters.
+
+## Format[#format]
+
+The `format` keyword allows for basic semantic identification of certain
+kinds of values that are commonly used. or user could define its own format and use them alongside
```
Comment (kakabisht) [inline pages/understanding-json-schema/reference/type.md:182]: Nitpicking: A typo + not in present tense
Suggestion: kinds of commonly used, or a user could define their format and use them alongside

File: pages/understanding-json-schema/reference/type.md [hunk 2]
```diff
@@ -175,3 +175,129 @@ types. For example, numeric types have a way of specifying a numeric
...
+
+For example, because JSON doesn\'t have a \"DateTime\" type, dates need to be encoded as strings.
+`format` allows the schema author to indicate that the string value
+should be interpreted as a date. By default, `format` is just an
+annotation and does not effect validation.
+
+Optionally, validator [implementations](../../learn/glossary#implementation) can provide a configuration option
+to enable `format` to function as an assertion rather than just an
+annotation. That means that validation will fail if, for example, a
```
Comment (kakabisht) [inline pages/understanding-json-schema/reference/type.md:191]: Nitpick :
Suggestion: validation fails when, for example a value with date format.

File: pages/understanding-json-schema/reference/type.md [hunk 3]
```diff
@@ -175,3 +175,129 @@ types. For example, numeric types have a way of specifying a numeric
...
+value with a `date` format isn\'t in a form that can be parsed as a
+date. This can allow values to be constrained beyond what the other
```
Comment (kakabisht) [inline pages/understanding-json-schema/reference/type.md:193]: Suggestion:
This allows values to be constrained beyond what the other tools in JSON Schema can do, including Regular Expressions.

File: pages/understanding-json-schema/reference/type.md [hunk 4]
```diff
@@ -175,3 +175,129 @@ types. For example, numeric types have a way of specifying a numeric
...
+
+> Implementations may provide validation for only a subset of the built-in
+> formats or do partial validation for a given format. For example, some
+> implementations may consider a string an email if it contains a `@`,
+> while others might do additional checks for other aspects of a well
+> formed email address.
+
+There is a bias toward networking-related formats in the JSON Schema
+specification, most likely due to its heritage in web technologies.
+However, custom formats may also be used, as long as the parties
```
Comment (kakabisht) [inline pages/understanding-json-schema/reference/type.md:205]: Suggestion: 
However, custom formats may also be used if the parties exchanging the JSON documents share information about the custom format types

File: pages/understanding-json-schema/reference/type.md [hunk 5]
```diff
@@ -175,3 +175,129 @@ types. For example, numeric types have a way of specifying a numeric
...
+It should be noted that `format` is not limited because it only defines a specific set of valid values for formats. Users may define their own custom keywords to work with any specific data type, such as `integer`, `double`, `float`, etc. Below, we cover the commonly used `string` formats specified in the JSON Schema specification.
+
+#### Dates and times
+
+Dates and times are represented in [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6). This is a subset
+of the date format also commonly known as [ISO8601 format](https://www.iso.org/iso-8601-date-and-time-format.html).
+
+- `"date-time"`: Date and time together, for example,
+  `2018-11-13T20:20:39+00:00`.
+- `"time"`: <StarInline label="New in draft 7" /> Time, for example, `20:20:39+00:00`
```
Comment (kakabisht) [inline pages/understanding-json-schema/reference/type.md:221]:
```suggestion
- `"time"`: <StarInline label="New in draft 7" /> Time, for example, `20:20:39+00:00`. 
```

File: pages/understanding-json-schema/reference/type.md (hunk 4 above)
Comment (kakabisht) [inline pages/understanding-json-schema/reference/type.md:205]: Suggestion: 
However, custom formats may also be used if the parties exchanging the JSON documents share information about the custom format types

File: pages/understanding-json-schema/reference/type.md [hunk 6]
```diff
@@ -175,3 +175,129 @@ types. For example, numeric types have a way of specifying a numeric
...
+  already know what a URI Template is, you probably don\'t need this
+  value.
+
+<Keywords label="single: json-pointer single: relative-json-pointer single: format; json-pointer single: format; relative-json-pointer" />
+
+#### JSON Pointer
+
+- `"json-pointer"`: <StarInline label="New in draft 6" /> A JSON Pointer, according to
+  [RFC6901](https://tools.ietf.org/html/rfc6901). There is more
+  discussion on the use of JSON Pointer within JSON Schema in
```
Comment (kakabisht) [inline pages/understanding-json-schema/reference/type.md:287]:
```suggestion
  discussion on using JSON Pointer within JSON Schema in
```

File: pages/understanding-json-schema/reference/type.md [hunk 7]
```diff
@@ -175,3 +175,129 @@ types. For example, numeric types have a way of specifying a numeric
...
+  the entire string contains only JSON Pointer content, e.g.
+  `/foo/bar`. JSON Pointer URI fragments, e.g. `#/foo/bar/` should use
+  `"uri-reference"`.
+- `"relative-json-pointer"`: <StarInline label="New in draft 7" /> A [relative JSON pointer](https://tools.ietf.org/html/draft-handrews-relative-json-pointer-01).
+
+<Keywords label="single: regex single: format; regex" />
+
+#### Regular Expressions
+
+- `"regex"`: <StarInline label="New in draft 7" /> A regular expression, which should be valid according to
```
Comment (kakabisht) [inline pages/understanding-json-schema/reference/type.md:298]:
```suggestion
- `"regex"`: <StarInline label="New in draft 7" /> A regular expression that should be valid according to
```

**Comment by kakabisht** [comment]
Hello, 
I followed the Microsoft style guide for technical writing to review the content. 
My only concern is formatting the code snippets. If you are fixing the code snippet indentation here, It would help if you also fixed the indentation across the entire documentation.

File: pages/understanding-json-schema/reference/type.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gregsdennis) [reply pages/understanding-json-schema/reference/type.md:182]: Grammatically, this still seems incomplete.

As written:

> The `format` keyword allows for basic semantic identification of certain
kinds of values that are commonly used. or user could define its own format and use them alongside

- "... commonly used. or user could ..." - The second sentence seems to start in the middle of a thought.
- "and use them alongside" - Alongside what?  This needs an object.

As suggested:

> The `format` keyword allows for basic semantic identification of certain kinds of commonly used, or a user could define their format and use them alongside

- "certain kinds of commonly used..." - Commonly used what?  Object needed.
- "and use them alongside" - Alongside what?  This needs an object.

---

My suggestion:

> The `format` keyword conveys semantic information for values that may be difficult or impossible to describe using JSON Schema. Typically, this semantic information is described by other documents. The JSON Schema Validation specification defines a number of formats, however  this keyword also allows schema authors to define their own formats.

**Review by gregsdennis — CHANGES_REQUESTED** [review]
@bhavukkalra please address the outstanding comments.  We can merge afterward.

**Comment by benjagm** [comment]
This issue is going to be closed due to inactivity.

**Pull request closed without merging**